server/task_executor_database_backup.go

```diff
@@ -88,18 +88,45 @@ func (*DatabaseBackupTaskExecutor) backupDatabase(ctx context.Context, server *S
 	}
 	defer driver.Close(ctx)
 
-	f, err := os.Create(filepath.Join(server.profile.DataDir, backup.Path))
+	backupFilePathLocal := filepath.Join(server.profile.DataDir, backup.Path)
+	backupFile, err := os.Create(backupFilePathLocal)
 	if err != nil {
 		return "", errors.Errorf("failed to open backup path: %s", backup.Path)
 	}
-	defer f.Close()
+	defer backupFile.Close()
```

Comment:
I may wrap the backupFile, err := os.Create(backupFilePathLocal), dump, defer backupFile.Close() in a separate method. So the backup file creation is separate from the upload

Otherwise, the double defer is less readable. Also we don't need to worry about deleting a file while there is an open handle on the just created file is OK or not.